packages/custom-url/src/Infrastructure/Sulu/Trash/CustomUrlTrashSubscriber.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\CustomUrl\Infrastructure\Sulu\Trash;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Sulu\Bundle\TrashBundle\Application\TrashManager\TrashManagerInterface;
+use Sulu\Component\DocumentManager\Event\ClearEvent;
+use Sulu\Component\DocumentManager\Event\FlushEvent;
+use Sulu\Component\DocumentManager\Event\RemoveEvent;
+use Sulu\Component\DocumentManager\Events;
+use Sulu\CustomUrl\Domain\Model\CustomUrl;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * @internal
+ */
+final class CustomUrlTrashSubscriber implements EventSubscriberInterface
```

Comment:
This is nothing CustomUrl specific. Maybe this makes sense to move to the trash bundle itself.